src/Admin/UserActionHistoryAdmin.php

```diff
@@ -0,0 +1,94 @@
+use Sonata\AdminBundle\Form\Type\ModelAutocompleteType;
+use Sonata\AdminBundle\Route\RouteCollectionInterface;
+use Sonata\DoctrineORMAdminBundle\Filter\ModelFilter;
+
+class UserActionHistoryAdmin extends AbstractAdmin
+{
+    protected function configureDefaultSortValues(array &$sortValues): void
+    {
+        parent::configureDefaultSortValues($sortValues);
+
+        $sortValues[DatagridInterface::SORT_BY] = 'date';
+        $sortValues[DatagridInterface::SORT_ORDER] = 'DESC';
+    }
+
+    protected function configureRoutes(RouteCollectionInterface $collection): void
+    {
+        $collection->clearExcept(['list']);
+    }
+
+    protected function configureDatagridFilters(DatagridMapper $filter): void
+    {
+        $filter
+            ->add('adherent', ModelFilter::class, [
+                'label' => 'Adhérent',
+                'show_filter' => true,
+                'field_type' => ModelAutocompleteType::class,
+                'field_options' => [
+                    'model_manager' => $this->getModelManager(),
+                    'property' => [
+                        'search',
+                    ],
+                    'to_string_callback' => static function (Adherent $adherent): string {
+                        return \sprintf(
+                            '%s (%s) [%s]',
+                            $adherent->getFullName(),
+                            $adherent->getEmailAddress(),
+                            $adherent->getId()
+                        );
+                    },
+                ],
+            ])
+            ->add('impersonificator', ModelFilter::class, [
+                'label' => 'Administrateur',
+                'show_filter' => true,
+                'field_type' => ModelAutocompleteType::class,
+                'field_options' => [
+                    'model_manager' => $this->getModelManager(),
+                    'property' => [
+                        'emailAddress',
+                    ],
+                ],
+            ])
+        ;
+    }
+
+    protected function configureListFields(ListMapper $list): void
+    {
+        $list
+            ->add('adherent', null, [
+                'label' => 'Adhérent',
+                'template' => 'admin/user_action_history/list_adherent.html.twig',
+            ])
+            ->add('type', null, [
+                'label' => 'Type',
+                'template' => 'admin/user_action_history/list_type.html.twig',
+            ])
+            ->add('date', null, [
+                'label' => 'Date',
+            ])
+            ->add('data', null, [
+                'label' => 'Données',
+            ])
+            ->add('impersonificator', null, [
+                'label' => 'Administrateur',
+            ])
+            ->add(ListMapper::NAME_ACTIONS, null, [
+                'virtual_field' => true,
+                'actions' => [
+                    'edit' => [],
+                    'delete' => [],
+                ],
+            ])
```

Comment:
you enable only list route in this Admin : `$collection->clearExcept(['list']);`